snap/info.go

```diff
@@ -356,7 +356,7 @@ type Info struct {
 	Plugs            map[string]*PlugInfo
 	Slots            map[string]*SlotInfo
 
-	Components map[string]Component
+	Components map[string]*Component
```

Comment:
I changed this to a pointer so that HookInfo can have a reference back to the component that it might have come from.